Mage.Sets/src/mage/cards/f/FestivalOfEmbers.java

```diff
@@ -0,0 +1,133 @@
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.abilities.effects.common.SacrificeSourceEffect;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.events.ZoneChangeEvent;
+import mage.players.Player;
+
+import java.util.UUID;
+
+/**
+ * @author notgreat
+ */
+public final class FestivalOfEmbers extends CardImpl {
+
+    public FestivalOfEmbers(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{4}{R}");
+
+
+        // During your turn, you may cast instant and sorcery spells from your graveyard by paying 1 life in addition to their other costs.
+        this.addAbility(new SimpleStaticAbility(new FestivalOfEmbersCastEffect()));
+
+        // If a card or token would be put into your graveyard from anywhere, exile it instead.
+        this.addAbility(new SimpleStaticAbility(new FestivalOfEmbersReplacementEffect()));
+
+        // {1}{R}: Sacrifice Festival of Embers.
+        this.addAbility(new SimpleActivatedAbility(new SacrificeSourceEffect(), new ManaCostsImpl<>("{1}{R}")));
+    }
+
+    private FestivalOfEmbers(final FestivalOfEmbers card) {
+        super(card);
+    }
+
+    @Override
+    public FestivalOfEmbers copy() {
+        return new FestivalOfEmbers(this);
+    }
+}
+
+//Based on Osteomancer Adept
+class FestivalOfEmbersCastEffect extends AsThoughEffectImpl {
+
+    FestivalOfEmbersCastEffect() {
+        super(AsThoughEffectType.CAST_FROM_NOT_OWN_HAND_ZONE, Duration.WhileOnBattlefield, Outcome.AIDontUseIt);
+        staticText = "During your turn, you may cast instant and sorcery spells from your graveyard by paying 1 life in addition to their other costs.";
+    }
+
+    private FestivalOfEmbersCastEffect(final FestivalOfEmbersCastEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return true;
+    }
+
+    @Override
+    public FestivalOfEmbersCastEffect copy() {
+        return new FestivalOfEmbersCastEffect(this);
+    }
+
+    @Override
+    public boolean applies(UUID objectId, Ability source, UUID affectedControllerId, Game game) {
+        if (!source.isControlledBy(affectedControllerId)) {
+            return false;
+        }
+        Card card = game.getCard(objectId);
+        Player player = game.getPlayer(affectedControllerId);
+        if (card == null
+                || player == null
+                || !game.getActivePlayerId().equals(affectedControllerId)
+                || !card.isOwnedBy(affectedControllerId)
+                || !card.isInstantOrSorcery(game)
+                || !game.getState().getZone(objectId).match(Zone.GRAVEYARD)) {
+            return false;
+        }
+        Costs<Cost> newCosts = new CostsImpl<>();
+        newCosts.addAll(card.getSpellAbility().getCosts());
```

Comment:
hm, what about if there's an alternate cost? e.g. overload

(not necessarily in scope for this PR if it turns out to be a whole thing)

Reply:
Looks like all the [other cards with similar text](https://scryfall.com/search?q=o%3A%2Fmay+cast+%28%3F%21%7E%29.*+other+cost%2F+f%3Av&unique=cards&as=grid&order=name) would have the same problem, seems out of scope to me.

Reply:
Okay, tracking by #12926